src/core/serialization.rs

```diff
@@ -132,7 +138,19 @@ impl Serializable for MasterSecretKey {
                 (sk_i, R25519PrivateKey::try_from_bytes(x_i)?),
             );
         }
-        Ok(Self { s, s1, s2, subkeys })
+
+        let kmac_key = match de.read_array::<{ SYM_KEY_LENGTH }>() {
+            Ok(key_bytes) => Some(SymmetricKey::try_from_bytes(key_bytes)?),
+            Err(_) => None,
+        };
```

Comment:
This could have been written:
```rust
        let kmac_key = de
            .read_array::<{ SYM_KEY_LENGTH }>()
            .ok()
            .map(SymmetricKey::try_from_bytes)
            .transpose()?;
```
But your version is fine.